### PR TITLE
RDKTV-31210 RDKTV-30775 RDKTV-31401 - Miracast Fixes

### PR DESCRIPTION
**Pull request by yuvaramachandran-gurusamy**
RDKTV-31210 RDKTV-30775 RDKTV-31401 - Miracast Fixes

Reason for change: Fixes to Disable and Enable Miracast discovery later while wakeup from Deepsleep and switching from one ssid to another ssid.
Test procedure: Miracast Functionality should work
Risks: High
Signed-off-by: yuvaramachandran_gurusamy <yuvaramachandran_gurusamy@comcast.com>

### CODE REVIEW

File: Miracast/MiracastService/P2P/MiracastP2P.cpp [hunk 1]
```diff
@@ -458,10 +458,21 @@ MiracastError MiracastP2P::discover_devices(void)
 {
     MiracastError ret = MIRACAST_FAIL;
     std::string command, retBuffer;
+    std::string opt_flag_buffer = MiracastCommon::parse_opt_flag("/opt/miracast_custom_p2p_scan");
```
Comment (apatel859) [inline Miracast/MiracastService/P2P/MiracastP2P.cpp:461]: do we need this in release?

File: Miracast/MiracastService/MiracastController.cpp [hunk 2]
```diff
@@ -1422,6 +1432,18 @@ void MiracastController::switch_launch_request_context(std::string& source_dev_i
     MIRACASTLOG_TRACE("Exiting...");
 }
 
+void MiracastController::enable_discovery_via_pwrmgr(void)
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastController.cpp:1435]: where this called?

File: Miracast/MiracastService/MiracastService.cpp [hunk 3]
```diff
@@ -59,13 +60,34 @@ using namespace std;
...
+	DEVICE_WIFI_STATE_UNINSTALLED = 0,
+	DEVICE_WIFI_STATE_DISABLED = 1,
+	DEVICE_WIFI_STATE_DISCONNECTED = 2,
+	DEVICE_WIFI_STATE_PAIRING = 3,
+	DEVICE_WIFI_STATE_CONNECTING = 4,
+	DEVICE_WIFI_STATE_CONNECTED = 5,
+	DEVICE_WIFI_STATE_FAILED = 6
+};
+
+static int32_t m_powerState = DEVICE_POWER_STATE_OFF;
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastService.cpp:87]: use DevicePowerStates instead int32

File: Miracast/MiracastService/MiracastService.cpp [hunk 4]
```diff
@@ -135,8 +173,92 @@ namespace WPEFramework
...
+				IARM_Result_t res;
+				IARM_CHECK( IARM_Bus_RemoveEventHandler(IARM_BUS_PWRMGR_NAME,IARM_BUS_PWRMGR_EVENT_MODECHANGED, pwrMgrModeChangeEventHandler) );
+			}
+		}
+
+		void MiracastService::pwrMgrModeChangeEventHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len)
+		{
+			if (nullptr == _instance)
+			{
+				LOGERR("#### MCAST-TRIAGE-NOK-PWR Miracast Discovery not enabled yet ####");
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastService.cpp:211]: fix comment

File: Miracast/MiracastService/MiracastService.cpp [hunk 5]
```diff
@@ -135,8 +173,92 @@ namespace WPEFramework
...
+				return;
+			}
+
+			if (0 == strcmp(owner, IARM_BUS_PWRMGR_NAME))
+			{
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastService.cpp:216]: combined with above if

Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastService/MiracastService.cpp:193]: Done

**Review by apatel859 — APPROVED** [review]

**Review by apatel859 — APPROVED** [review]

**Pull request merged**